### PR TITLE
OCPBUGS-43499: Add openstack provision etcd slow profile step

### PR DESCRIPTION
**Pull request by MiguelCarpio**
Set ETCD Slow profile: This profile is optimized for infrastructure with high variability on network or storage latency.

The openstack-provision-etcd-disk-speed step will patch the etcd cluster to allow slower disks.

This PR will focus on the following ShiftStack periodic jobs:

- e2e-openstack-ovn-upgrade (Without etcd-on-ramfs)
- e2e-openstack-ovn-etcd-scaling (Without etcd-on-ramfs)
- e2e-openstack-ovn-techpreview (With etcd-on-ramfs)
- e2e-openstack-csi-cinder (With etcd-on-ramfs)
- e2e-openstack-csi-manila (With etcd-on-ramfs)

### CODE REVIEW

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-techpreview-e2e-openstack-ovn-techpreview

File: ci-operator/step-registry/openstack/provision/etcd-disk-speed/openstack-provision-etcd-disk-speed-commands.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+function wait_for_etcd_patch_done() {
+  info 'INFO: Waiting for patch etcd cluster...'
+  if
+    ! oc wait --timeout=30m --for=jsonpath='{.spec.controlPlaneHardwareSpeed}'="Slower" etcd cluster 1>/dev/null || \
+    ! oc wait --timeout=30m --for=jsonpath='{.status.conditions[0].status}'="False" co etcd 1>/dev/null || \
+    ! oc wait --timeout=30m --for=jsonpath='{.status.conditions[1].status}'="False" co etcd 1>/dev/null || \
+    ! oc wait --timeout=30m --for=jsonpath='{.status.conditions[2].status}'="True" co etcd 1>/dev/null; then
+      info "ERROR: Etcd patch failed..."
```
Comment (EmilienM) [inline ci-operator/step-registry/openstack/provision/etcd-disk-speed/openstack-provision-etcd-disk-speed-commands.sh:20]: Waiting for conditions by their list position isn't ideal. Another way is to run:

```bash
for i in {1..30}; do
  ETCD_CO_AVAILABLE=$(oc get co etcd | grep etcd | awk '{print $3}')
  if [[ "${ETCD_CO_AVAILABLE}" == "True" ]]; then
    echo "Patched successfully!"
    break
  fi
  sleep 15
done
if [[ "${ETCD_CO_AVAILABLE}" != "True" ]]; then
  echo "Etcd patch failed..."
  exit 1
fi
```

File: ci-operator/config/shiftstack/ci/shiftstack-ci-release-4.17__techpreview.yaml [hunk 2]
```diff
@@ -44,6 +44,7 @@ tests:
   steps:
     cluster_profile: openstack-vexxhost
     env:
+      ETCD_DISK_SPEED: slow
```
Comment (EmilienM) [inline ci-operator/config/shiftstack/ci/shiftstack-ci-release-4.17__techpreview.yaml:47]: please enable it in more jobs than just this one. We really want to see if that makes a difference. If possible jobs from multiple cluster profiles (CSI jobs, etc).

Comment (MiguelCarpio) [reply ci-operator/config/shiftstack/ci/shiftstack-ci-release-4.17__techpreview.yaml:47]: I agree. I didn't want to make this PR bigger, but I prefer to test it with more jobs.

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-dualstack periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-dualstack-v6primary periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-ovn-etcd-scaling

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-ovn-etcd-scaling periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-ovn-etcd-scaling

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-csi-cinder periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-csi-cinder periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-csi-cinder

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-csi-cinder

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-csi-cinder

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-csi-cinder periodic-ci-shiftstack-ci-release-4.17-upgrade-from-stable-4.16-e2e-openstack-ovn-upgrade periodic-ci-shiftstack-ci-release-4.18-upgrade-from-stable-4.17-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-csi-cinder

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-ovn-etcd-scaling periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-ovn-etcd-scaling

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-csi-cinder

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-upgrade-from-stable-4.16-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-csi-manila periodic-ci-shiftstack-ci-release-4.18-upgrade-from-stable-4.17-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-csi-manila periodic-ci-shiftstack-ci-release-4.18-upgrade-from-stable-4.17-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-upgrade-from-stable-4.16-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-upgrade-from-stable-4.16-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-upgrade-from-stable-4.16-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-upgrade-from-stable-4.16-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-upgrade-from-stable-4.15-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-upgrade-from-stable-4.15-e2e-openstack-ovn-upgrade

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-upgrade-from-stable-4.15-e2e-openstack-ovn-upgrade

**Pull request closed without merging**